#! /usr/bin/env python

PACKAGE='kcl_sbc'

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()
#       Name       Type      Level Description     Default Min   Max
gen.add("f_ref", double_t,    0,    "Force Reference", 0, 0.0,	5)
gen.add("p_gain", double_t,    0,    "P Gain", 0.35, 0.0,	2)
gen.add("i_gain",       double_t,    0,    "I Gain", 0.02,     0, 0.2)
gen.add("d_gain",       double_t,    0,    "D Gain", 0.02,     0, 0.2)
gen.add("deadband",       double_t,    0,    "Deadband (%)", 5,     0, 100)
gen.add("a_windup",       double_t,    0,    "Anti-Windup (%)", 10,     0, 100)


#size_enum = gen.enum([ gen.const("Small",      int_t, 0, "A small constant"),
#                       gen.const("Medium",     int_t, 1, "A medium constant"),
#                       gen.const("Large",      int_t, 2, "A large constant"),
#                       gen.const("ExtraLarge", int_t, 3, "An extra large constant")],
#                     "An enum to set size")
#gen.add("size", int_t, 0, "A size parameter which is edited via an enum", 1, 0, 3, edit_method=size_enum)

exit(gen.generate(PACKAGE, "kcl_sbc", "PID_params"))
